models/deepseek_v4_flash_mtp/sample.py

```
@@ -0,0 +1,628 @@
+            tail_max_scores[0 : VOCAB // TOPK_ROW_WIDTH_TILE % SAMPLE_BLOCK_ROWS_TILE, :] = tail_scores
+            tail_minima = pl.row_min(tail_min_scores)
+            tail_maxima = pl.row_max(tail_max_scores)
+            for tail_lane in pl.range(VOCAB // TOPK_ROW_WIDTH_TILE % SAMPLE_BLOCK_ROWS_TILE):
+                tail_minimum = pl.read(tail_minima, [tail_lane, 0])
+                tail_maximum = pl.read(tail_maxima, [tail_lane, 0])
+                if tail_minimum < search_lower:
+                    search_lower = tail_minimum
+                if tail_maximum > search_upper:
+                    search_upper = tail_maximum
+
+            for search_step in pl.range(TOPK_SEARCH_STEPS):
```

Ruff (0.16.2)

[warning] Loop control variable `search_step` not used within loop body

Rename unused `search_step` to `_search_step`

(B007)